src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests_ListPatterns.cs

```diff
@@ -2412,6 +2412,40 @@ void Test(int[][] a)
                 Diagnostic(ErrorCode.ERR_SwitchCaseSubsumed, "[[42]]").WithLocation(9, 18));
         }
 
+        [Fact(Skip = "TODO")]
+        public void Subsumption_16()
```

Comment:
🚧 We're relating input chains too aggressively. I'll need to investigate more to see where should we stop.

Reply:
The fix was simpler than I thought. We had dropped the next node in dag lowering.